### ISSUE TITLE
All hamcrest tests fail

### ISSUE THREAD

**Issue by Kombustor** [se2p/programming-styles-ss-2020-public-tests#14]
All of the tests in "TrickyInputTest.java" fail for me. I'm using the correct symbols pasted from the documentation, and I have no issues with the other tests by @mhrimaz for example.

The output I'm getting is:
```
There were 4 failures:
1) testThatGivenTrickyWinningInputsTheProgramOutputsCorrectly(TrickyInputTest)
java.lang.AssertionError: 
Expected: [╔════════╤═══════╗, ║  ┌──┐1 │   2   ║, ║  │ 3│  │       ║, ║  │45│  │       ║, ║  └──┘6 │       ║, ╚════════╧═══════╝]
     but: was ["╔════════╤═══╤═══════╗", "║  ┌──┐1 │   │   2   ║", "║  │  │  │ 3 │       ║", "║  │45│  │   │       ║", "║  └──┘6 │   │       ║", "╚════════╧═══╧═══════╝"]
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at TrickyInputTest.callMatchersForFinalBoardAndMessage(TrickyInputTest.java:72)
	at TrickyInputTest.testThatGivenTrickyWinningInputsTheProgramOutputsCorrectly(TrickyInputTest.java:230)
2) testThatGivenTrickyLosingInputsTheProgramOutputsCorrectly(TrickyInputTest)
java.lang.AssertionError: 
Expected: [╔════════╤═══╤═══╤═══════╗, ║  ┌──┐1 │   │   │   2   ║, ║  │  │  │   │   │   3   ║, ║  │4 │  │   │   │       ║, ║  └──┘6 │   │   │   5   ║, ╚════════╧═══╧═══╧═══════╝]
     but: was ["╔════════╤═══╤═══╤═══════╗", "║  ┌──┐1 │   │   │   2   ║", "║  │  │  │   │   │   3   ║", "║  │4 │  │   │   │       ║", "║  └──┘6 │   │ 5 │       ║", "╚════════╧═══╧═══╧═══════╝"]
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at TrickyInputTest.callMatchersForFinalBoardAndMessage(TrickyInputTest.java:72)
	at TrickyInputTest.testThatGivenTrickyLosingInputsTheProgramOutputsCorrectly(TrickyInputTest.java:183)
3) testThatGivenTrickyDrawingInputsTheProgramOutputsCorrectly(TrickyInputTest)
java.lang.AssertionError: 
Expected: [╔════════╤═══╤═══════╗, ║  ┌──┐1 │   │   2   ║, ║  │  │  │   │   3   ║, ║  │45│  │   │       ║, ║  └──┘6 │   │       ║, ╚════════╧═══╧═══════╝]
     but: was ["╔════════╤═══╤═══╤═══════╗", "║  ┌──┐1 │   │   │   2   ║", "║  │  │  │   │   │   3   ║", "║  │4 │  │   │   │       ║", "║  └──┘6 │ 5 │   │       ║", "╚════════╧═══╧═══╧═══════╝"]
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at TrickyInputTest.callMatchersForFinalBoardAndMessage(TrickyInputTest.java:72)
	at TrickyInputTest.testThatGivenTrickyDrawingInputsTheProgramOutputsCorrectly(TrickyInputTest.java:283)
4) testThatGivenTrickyInputsTheProgramOutputsCorrectly(TrickyInputTest)
java.lang.AssertionError: 
Expected: [╔════════╤═══╤═══╤═══════╗, ┌─────────────────┐, │ Missing inputs. │, │ The games ends! │, └─────────────────┘, ╚════════╧═══╧═══╧═══════╝]
     but: was [" ╔════════╤═══╤═══╤═══════╗", "┌──────────────────────────┐", "│      Missing inputs.     │", "│      The games ends!     │", "└──────────────────────────┘", " ╚════════╧═══╧═══╧═══════╝"]
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at TrickyInputTest.callMatchersForFinalBoardAndMessage(TrickyInputTest.java:69)
	at TrickyInputTest.testThatGivenTrickyInputsTheProgramOutputsCorrectly(TrickyInputTest.java:125)
```

I'm not sure if the hamcrest tests are even required anymore when #12 (and @mhrimaz's javascript implementation of it) will be merged? @alessiogambi 

@stocker4141

**Comment by mhrimaz** [comment]
@Kombustor if you have time to write JS tests, go for it 😄 . Otherwise probably I will do that after weekend.

**Comment by Kombustor** [comment]
@mhrimaz done in #12

**Comment by alessiogambi** [comment]
The first test fails because the position of fish 3 is different. Not sure whether this is a problem in the tests or in the code. Mayne some better assertions can help debugging? Something that shows where the error is a bit more precisely...

**Comment by Kombustor** [comment]
But is the effort debugging it worth when #12 can be merged? 
I respect @stocker4141's work, it's impressive, but it's so much more overhead and effort creating a test compared to the input/output files.

**Comment by alessiogambi** [comment]
Don't get me wrong. Input/output test are useful. the problem is that they might check too much at the same time. Meaning that when they fail is not necessarily easy to understand why the tests failed. And that's exactly what you need for debugging... A clear indication on where the bug is located. We will keep the input output test around and use them to test students solution  nevertheless

**Comment by Kombustor** [comment]
> Don't get me wrong. Input/output test are useful. the problem is that they might check too much at the same time. Meaning that when they fail is not necessarily easy to understand why the tests failed. And that's exactly what you need for debugging... A clear indication on where the bug is located. We will keep the input output test around and use them to test students solution nevertheless

That perfectly makes sense, I didn't think about that in the first place

**Comment by stocker4141** [comment]
Sorry for only responding now: 
The issue is that the test check the wrong board. Intended was to check the board before the message box is printed but infact the board that is under the box is checked.
Pull request with a fix coming soon.

**Comment by alessiogambi** [comment]
I consider this closed by #17
